security/auth.go

```diff
@@ -139,3 +139,25 @@ func UserAuthHook(insecureMode bool, tlsState *tls.ConnectionState) (
 		return nil
 	}, nil
 }
+
+// CheckRequestUser extracts user from the request and verify.
```

Comment:
I've actually just removed this completely. `rpc.Send` is only ever used from the `kv` package which itself is a non-public interface. We don't need to check security again. It's on my TODO list to move `rpc.Send` to the `kv` package to make this clearer.